pkg/kv/kvserver/rangefeed/budget.go

```diff
@@ -16,11 +16,21 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/rpc"
+	"github.com/cockroachdb/cockroach/pkg/util/envutil"
 	"github.com/cockroachdb/cockroach/pkg/util/mon"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/errors"
 )
 
+// Environment variable controlling rangefeed budgets. If false budgets are
+// disabled otherwise enabled.
+const rangeFeedBudgetEnvVarName = "COCKROACH_USE_RANGEFEED_MEM_BUDGETS"
+
+// Pre allocated memory limit for system rangefeeds.
+const systemRangeFeedLimit = int64(64 * 1024 * 1024)
```

Comment:
nit: consider calling this budget rather than limit. The primary motivation here isn't so much to limit the amount of memory used by rangefeeds as it is to have a reserved budget for them. In fact, it is a bit unfortunate that this can't borrow from the main rangefeed budget, but I suppose that's for another day.

Anyway, this should be configurable, at the very least through an environment variable, in case it ends up causing problems.